proto/redpanda/api/dataplane/v1/mcp.proto

```diff
@@ -0,0 +1,664 @@
+      description: "Stop a running Redpanda Connect MCP server."
+      responses: {
+        key: "200"
+        value: {
+          description: "Stopped"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1.MCPServer"}
+          }
+        }
+      }
+      responses: {
+        key: "404"
+        value: {
+          description: "Not Found"
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+    option (redpanda.api.auth.v1.authorization) = {
+      required_permission: PERMISSION_EDIT
+      api: API_REDPANDA_MCP_SERVER
+    };
+  }
+
+  // StartMCPServer starts a specific Redpanda Connect MCP server that has been previously stopped.
+  rpc StartMCPServer(StartMCPServerRequest) returns (StartMCPServerResponse) {
+    option (google.api.http) = {post: "/v1/redpanda-connect/mcp-servers/{id}:start"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Start a Redpanda Connect MCP server"
+      description: "Start a stopped Redpanda Connect MCP server."
+      responses: {
+        key: "200"
+        value: {
+          description: "Started"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1.MCPServer"}
+          }
+        }
+      }
+      responses: {
+        key: "404"
+        value: {
+          description: "Not Found"
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+    option (redpanda.api.auth.v1.authorization) = {
+      required_permission: PERMISSION_EDIT
+      api: API_REDPANDA_MCP_SERVER
+    };
+  }
+
+  // The configuration schema includes available components and processors in this Redpanda Connect MCP Server instance.
+  rpc GetMCPServerServiceConfigSchema(GetMCPServerServiceConfigSchemaRequest) returns (GetMCPServerServiceConfigSchemaResponse) {
+    option (google.api.http) = {get: "/v1/redpanda-connect/mcp-config-schema"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Retrieve the schema for Redpanda Connect MCP Servers configurations."
+      description: "The configuration schema includes available components and processors in this Redpanda Connect MCP Server instance."
+      responses: {
+        key: "200"
+        value: {
+          description: "OK"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1.GetMCPServerServiceConfigSchemaResponse"}
+          }
+        }
+      }
+    };
+    option (redpanda.api.auth.v1.authorization) = {
+      required_permission: PERMISSION_VIEW
+      api: API_REDPANDA_MCP_SERVER
+    };
+  }
+
+  // GetMCPServersForSecret implements the get MCP servers for secret method which lists the MCP servers
+  // in the Redpanda cluster for the given secret.
+  rpc GetMCPServersForSecret(GetMCPServersForSecretRequest) returns (GetMCPServersForSecretResponse) {
```

Comment:
do we need this one?

Reply:
Hmm we were using this endpoint for pipelines to lookup whether a secret is still in use and warn the user when deleting. Not sure if we need it or whether we would still be able to retrieve all pipelines including MCP servers when calling that endpoint on the pipelines endpoint.